### DISCUSSION TITLE
How to generate only model-related types without client?

### DISCUSSION THREAD

**Discussion by StringKe** [prisma/prisma#16316] [Q&A] [answered]
As much as I need `model`,`enum` related type definitions, I need to give them to the front end to use.

At the moment I need to copy the relevant code to a new file myself each time, does `prisma-client-js` have an option to control that only the type definition is generated?

**Comment by nurul3101** [comment, marked as answer]
Hello @StringKe 

The type definitions for the models and enums are created by Prisma when you invoke `npx prisma generate` command, and you can directly import them from generated PrismaClient. To share these types with frontend you can create an npm package which just exports these generated types.

For reference, here is another [GitHub Discussion](https://github.com/prisma/prisma/discussions/5526) which discusses this same topic.

**Reply by StringKe** [reply]
The `npx prisma generate` generation contains `@prisma/client` related types, but I only need `enum` and `model` and don't need other type definitions for the client.

**Reply by pantharshit00** [reply]
In that case, please create a feature request here: https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=

Right now there is no way to configure the generated output.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
